### PR TITLE
Release mo dev guide refactoring

### PR DESCRIPTION
**Pull request by lazarevevgeny**
This PR contains the following MO documentation changes:

- big new document describing:
  - model representation in memory
  - model conversion pipeline
  - describing MO extensibility features
     - operation implementation
     - extractors implementation
     - various types of transformations implementation
- refactoring/cleaning old MO documentation related to extensibility
- completely re-written the "Custom layer guide" and add a real life example on how to enable the MRI model using custom CPU extension

### CODE REVIEW

**Comment by ilyachur** [comment]
Can you connect the Model Optimizer customization documentation with [Extensibility DG](https://github.com/openvinotoolkit/openvino/tree/master/docs/IE_DG/Extensibility_DG)?

File: docs/MO_DG/prepare_model/customize_model_optimizer/Extending_MXNet_Model_Optimizer_with_New_Primitives.md [hunk 1]
```diff
@@ -1,45 +1,41 @@
-# Extending the MXNet Model Optimizer with New Primitives  {#openvino_docs_MO_DG_prepare_model_customize_model_optimizer_Extending_MXNet_Model_Optimizer_with_New_Primitives}
+# Extending the Model Optimizer for Custom MXNet* Operations {#openvino_docs_MO_DG_prepare_model_customize_model_optimizer_Extending_MXNet_Model_Optimizer_with_New_Primitives}
 
-This section describes how you can create a Model Optimizer extension for a custom layer from your MXNet* model. It supplements the main document [Extending Model Optimizer with New Primitives](Extending_Model_Optimizer_with_New_Primitives.md) and provides a step-by-step procedure. To create an extension for a particular layer, perform the following steps:
+This section provides instruction on how to support a custom MXNet operation (or as it called in the MXNet documentation
+"operator" or "layer") which is not a part of the MXNet operation set. For example, if the operator is implemented using
```
Comment (evolosen) [inline docs/MO_DG/prepare_model/customize_model_optimizer/Extending_MXNet_Model_Optimizer_with_New_Primitives.md:4]: I don't  understand. "which is not a part of the MXNet operation set". Set of one operation or set of operations.

File: docs/MO_DG/prepare_model/customize_model_optimizer/Extending_MXNet_Model_Optimizer_with_New_Primitives.md [hunk 2]
```diff
@@ -1,45 +1,41 @@
...
+the following [guide](https://mxnet.apache.org/versions/1.7.0/api/faq/new_op.html)).
+
+This section describes a procedure on how to extract operator attributes in the Model Optimizer. The rest of the
```
Comment (evolosen) [inline docs/MO_DG/prepare_model/customize_model_optimizer/Extending_MXNet_Model_Optimizer_with_New_Primitives.md:7]: "extract operator attributes". may be "attributes' operator" or "operator of attributes" or "operator, attributes"

File: docs/MO_DG/prepare_model/customize_model_optimizer/Extending_MXNet_Model_Optimizer_with_New_Primitives.md [hunk 3]
```diff
@@ -1,45 +1,41 @@
...
+operation enabling pipeline and documentation on how to support MXNet operations from standard MXNet operation set is
+described in the main document [Customize_Model_Optimizer](Customize_Model_Optimizer.md).
+
+## Writing Extractor for Custom MXNet Operation
+Custom MXNet operations have an attribute `op` (defining the type of the operation) equal to `Custom` and attribute
+`op_type` which is an operation type defined by an user. In order to extract attributes for such an operation it is
```
Comment (evolosen) [inline docs/MO_DG/prepare_model/customize_model_optimizer/Extending_MXNet_Model_Optimizer_with_New_Primitives.md:13]: "for such an operation" may be for such operations or for such an operation('op_type')

Comment (evolosen) [reply docs/MO_DG/prepare_model/customize_model_optimizer/Extending_MXNet_Model_Optimizer_with_New_Primitives.md:13]: "it is". I think 'it' is not need here.

File: docs/MO_DG/prepare_model/customize_model_optimizer/Extending_MXNet_Model_Optimizer_with_New_Primitives.md (hunk 1 above)
Comment (evolosen) [inline docs/MO_DG/prepare_model/customize_model_optimizer/Extending_MXNet_Model_Optimizer_with_New_Primitives.md:4]: "if the operator is implemented using"  may be operation?

File: docs/MO_DG/prepare_model/customize_model_optimizer/Extending_Model_Optimizer_with_Caffe_Python_Layers.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+## Writing Extractor for Caffe Python Layer
+Custom Caffe Python layers have an attribute `type` (defining the type of the operation) equal to `Python` and two
+mandatory attributes `module` and `layer` in the `python_param` dictionary. The `module` defines the Python module name
+with the layer implementation, while `layer` value is an operation type defined by an user. In order to extract
+attributes for such an operation it is necessary to implement extractor class inherited from the
+`CaffePythonFrontExtractorOp` class instead of `FrontExtractorOp` class used for standard framework layers. The `op`
+class attribute value should be set to the `module + "." + layer` value so the extractor is triggered for this kind of
+operation.
+
+Here is a simplified example of the extractor for the custom operation Proposal from Faster-R-CNN model mentioned above.
```
Comment (evolosen) [inline docs/MO_DG/prepare_model/customize_model_optimizer/Extending_Model_Optimizer_with_Caffe_Python_Layers.md:37]: may be there is simplefiedexample ...

File: docs/MO_DG/prepare_model/customize_model_optimizer/Extending_MXNet_Model_Optimizer_with_New_Primitives.md (hunk 2 above)
Comment (evolosen) [reply docs/MO_DG/prepare_model/customize_model_optimizer/Extending_MXNet_Model_Optimizer_with_New_Primitives.md:7]: And I think you should change 'operator' to 'operation'.

File: docs/MO_DG/prepare_model/customize_model_optimizer/Extending_MXNet_Model_Optimizer_with_New_Primitives.md (hunk 1 above)
Comment (lazarevevgeny) [reply docs/MO_DG/prepare_model/customize_model_optimizer/Extending_MXNet_Model_Optimizer_with_New_Primitives.md:4]: Here we use the word "operator" as in the MXNet documentation so it is correct usage.

File: docs/MO_DG/prepare_model/customize_model_optimizer/Extending_MXNet_Model_Optimizer_with_New_Primitives.md (hunk 2 above)
Comment (lazarevevgeny) [reply docs/MO_DG/prepare_model/customize_model_optimizer/Extending_MXNet_Model_Optimizer_with_New_Primitives.md:7]: Again, we refer to the MXNet operator, not OpenVINO operation. 

The point of this sentence is that we have an operator (or operation) and we want to extract attributes of this operation. For example, for Convolution we want to extract strides and paddings etc. Both of your suggestions have different meaning.

File: docs/MO_DG/prepare_model/customize_model_optimizer/Extending_MXNet_Model_Optimizer_with_New_Primitives.md (hunk 3 above)
Comment (lazarevevgeny) [reply docs/MO_DG/prepare_model/customize_model_optimizer/Extending_MXNet_Model_Optimizer_with_New_Primitives.md]: Fixed.

**Review by andrew-zaytsev — APPROVED** [review]

**Comment by andrew-zaytsev** [comment]
Let's merge this version. The content will be proofread later.

**Pull request merged**